modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -447,6 +472,32 @@ private DataRegionConfiguration createMetastoreDataRegionConfig(DataStorageConfi
         return cfg;
     }
 
+    /** */
+    private DataRegionConfiguration createDefragmentationDataRegionConfig(long regionSize) {
```

Comment:
What are the drawbacks of such an approach? 
Will the region be deallocated when the defragmentation ends?

Reply:
They will be deallocated when node stops, just like all the other regions. And I don't think I understand the first question.
The thing is that we either create two new data regions dynamically or we deeply refactor and change our page memory classes, which would take many months and add inevitable performance degradation in everyday scenarios(

Reply:
Is it true that the defragmentation region allocated at the node start will not allow the node to allocate the full amount of available memory until node restart? The node may live months without restart with shrunk memory for technical needs, right?

Reply:
No, node must be restarted when defragmentation is completed. It's not part of the topology during this process, so everything's fine here.